### PR TITLE
Delete Trigger Feed Before Trigger

### PR DESCRIPTION
**Pull request by dubeejw**
- Dissociate a feed from a trigger before a trigger is deleted

### CODE REVIEW

**Comment by dubeejw** [comment]
Related to https://github.com/openwhisk/openwhisk-package-alarms/issues/55

**Comment by dubeejw** [comment]
@jasonpet, @bjustin-ibm, FYI.

**Comment by dubeejw** [comment]
PG2 1495

**Comment by dubeejw** [comment]
@mdeuser, please review.

**Comment by rabbah** [comment]
The deletion of the trigger before the feed is deliberate.

File: tools/cli/go-whisk-cli/commands/trigger.go [hunk 1]
```diff
@@ -400,9 +406,9 @@ var triggerDeleteCmd = &cobra.Command{
 
         client.Namespace = qName.namespace
 
-        retTrigger, _, err = client.Triggers.Delete(qName.entityName)
+        retTrigger, _, err = client.Triggers.Get(qName.entityName)
         if err != nil {
-            whisk.Debug(whisk.DbgError, "client.Triggers.Delete(%s) failed: %s\n", qName.entityName, err)
+            whisk.Debug(whisk.DbgError, "client.Triggers.Get(%s) failed: %s\n", qName.entityName, err)
             errStr := wski18n.T("Unable to delete trigger '{{.name}}': {{.err}}",
```
Comment (mdeuser) [inline tools/cli/go-whisk-cli/commands/trigger.go]: "Unable to delete trigger" -> "Unable to get trigger"

File: tools/cli/go-whisk-cli/commands/trigger.go [hunk 2]
```diff
@@ -415,26 +421,30 @@ var triggerDeleteCmd = &cobra.Command{
```
Comment (mdeuser) [inline tools/cli/go-whisk-cli/commands/trigger.go]: regarding lines
```
        // Get full feed name from trigger delete request as it is needed to delete the feed
        if retTrigger != nil && retTrigger.Annotations != nil {
```
what happens if retTrigger == nil or retTrigger.Annotations == nil (i.e. need to provide some `else` logic) ?

File: tools/cli/go-whisk-cli/commands/trigger.go [hunk 3]
```diff
@@ -415,26 +421,30 @@ var triggerDeleteCmd = &cobra.Command{
...
                 err = configureFeed(qName.entityName, fullFeedName)
                 if err != nil {
-                    whisk.Debug(whisk.DbgError, "configureFeed(%s, %s) failed: %s\n", qName.entityName, flags.common.feed,
-                        err)
-                    errStr := wski18n.T("Unable to delete trigger '{{.name}}': {{.err}}",
-                            map[string]interface{}{"name": qName.entityName, "err": err})
-                    werr := whisk.MakeWskErrorFromWskError(errors.New(errStr), err, whisk.EXITCODE_ERR_GENERAL, whisk.DISPLAY_MSG, whisk.NO_DISPLAY_USAGE)
-
-                    return werr
+                    whisk.Debug(whisk.DbgError, "configureFeed(%s, %s) failed: %s\n", qName.entityName, flags.common.feed, err)
```
Comment (mdeuser) [inline tools/cli/go-whisk-cli/commands/trigger.go]: flags.common.feed -> fullFeedName

**Comment by mdeuser** [comment]
@dubeejw - is there resolution re: @rabbah's comment above?

**Comment by rabbah** [comment]
Resolution: ttrigger is deleted whether the feed action succeeds or not.

**Comment by dubeejw** [comment]
PG4 182, approved. Need to make review changes still.

File: tools/cli/go-whisk-cli/commands/trigger.go (hunk 2 above)
Comment (dubeejw) [inline tools/cli/go-whisk-cli/commands/trigger.go]: That means no feed is associated with the trigger, so deletion of the feed is not carried out.

**Comment by dubeejw** [comment]
@mdeuser, PR updated per review comments.

**Comment by dubeejw** [comment]
@mdeuser, in regard to the concern above, the trigger will always be deleted even if the feed fails to delete.

**Comment by mdeuser** [comment]
@dubeejw - go changes look good.

How about https://github.com/openwhisk/openwhisk/pull/2187#issuecomment-297492641 ?

**Comment by dubeejw** [comment]
@mdeuser, see https://github.com/openwhisk/openwhisk/pull/2187#issuecomment-298384969.

**Pull request merged**